councilmatic_core/management/commands/convert_rtf.py

```diff
@@ -0,0 +1,137 @@
+        (3) updating the full_text field with new html.
+        '''
+        listener = subprocess.Popen(['unoconv', '--listener'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        try:
+            self.add_html()
+        finally:
+            listener.kill()
+
+    def get_rtf(self):
+        self.connection.execute("SET local timezone to '{}'".format(settings.TIME_ZONE))
+        with engine.begin() as connection:
+             
+            # Only apply this query to most recently updated (or created) bills.
+            max_updated = Bill.objects.all().aggregate(Max('ocd_updated_at'))['ocd_updated_at__max']
+            # if max_updated is None or options['update_all']:
+            if max_updated is None or self.update_all:
+                max_updated = datetime.datetime(1900, 1, 1)    
+
+            query = '''
+            SELECT ocd_id, full_text
+            FROM councilmatic_core_bill
+            WHERE updated_at >= '{}'
+            AND full_text is not null
+            ORDER BY updated_at DESC
+            '''.format(max_updated)
+
+            result = connection.execution_options(stream_results=True).execute(query)
+
+            yield from result
+
+    def convert_rtf(self):
+        rtf_results = self.get_rtf()
+
+        self.log_message('Converting RTF to HTML....')
+
+        inserts = ''
+        for bill_data in rtf_results:
+            ocd_id = bill_data['ocd_id']
+            rtf_string = bill_data['full_text']
+            
+            process = subprocess.run(['unoconv', '--stdin', '--stdout', '-f', 'html'], input=rtf_string.encode(), stdout=subprocess.PIPE, stderr=subprocess.DEVNULL, timeout=15)
+
+            html = process.stdout.decode('utf-8')
+
+            print('.', end='')
+            sys.stdout.flush()
+
+            yield html, ocd_id
+           
+
+    def add_html(self):
+        html_results = self.convert_rtf()
+
+        self.connection.execute("SET local timezone to '{}'".format(settings.TIME_ZONE))
+        query = '''
+            UPDATE councilmatic_core_bill as bills
+            SET full_text = :html
+            WHERE bills.ocd_id = :ocd_id  
+        '''
+
+        chunk = []
+        for html, ocd_id in html_results:
+          chunk.append({'html': html, 'ocd_id': ocd_id})
+          if len(chunk) == 1000:
+            with self.connection.begin() as trans:
+                self.connection.execute(sa.text(query), *chunk)
+                
+                chunk = []
+
+        # Update bills when less than 1,000 elements in a chunk. 
+        if chunk:
+            with self.connection.begin() as trans:
+                self.connection.execute(sa.text(query), *chunk)
+
+        self.log_message('Bills have valid, viewable HTML!',
+                         fancy=True,
+                         center=True,
+                         style='SUCCESS')
+            
+    def log_message(self,
```

Comment:
the standard logging module handles almost all of this for you?

Reply:
Yep, it does.